titus-server-master/src/main/java/com/netflix/titus/master/mesos/kubeapiserver/direct/DefaultKubeApiFacade.java

```diff
@@ -0,0 +1,243 @@
+                this.opportunisticResourceInformer = createOpportunisticResourceInformer(sharedInformerFactory);
+
+                sharedInformerFactory.startAllRegisteredInformers();
+
+                logger.info("Kube node and pod informers activated");
+            } catch (Exception e) {
+                logger.error("Could not initialize Kube client shared informer", e);
+                if (sharedInformerFactory != null) {
+                    ExceptionExt.silent(() -> sharedInformerFactory.stopAllRegisteredInformers());
+                }
+                sharedInformerFactory = null;
+                nodeInformer = null;
+                podInformer = null;
+                throw e;
+            }
+        }
+    }
+
+    private SharedIndexInformer<V1Node> createNodeInformer(SharedInformerFactory sharedInformerFactory) {
+        return sharedInformerFactory.sharedIndexInformerFor(
+                (CallGeneratorParams params) -> coreV1Api.listNodeCall(
+                        null,
+                        null,
+                        null,
+                        null,
+                        null,
+                        params.resourceVersion,
+                        params.timeoutSeconds,
+                        params.watch,
+                        null,
+                        null),
+                V1Node.class,
+                V1NodeList.class,
+                configuration.getKubeApiServerIntegratorRefreshIntervalMs()
+        );
+    }
+
+    private SharedIndexInformer<V1Pod> createPodInformer(SharedInformerFactory sharedInformerFactory) {
+        return sharedInformerFactory.sharedIndexInformerFor(
+                (CallGeneratorParams params) -> coreV1Api.listNamespacedPodCall(
+                        KUBERNETES_NAMESPACE,
+                        null,
+                        null,
+                        null,
+                        null,
+                        null,
+                        params.resourceVersion,
+                        params.timeoutSeconds,
+                        params.watch,
+                        null,
+                        null
+                ),
+                V1Pod.class,
+                V1PodList.class,
+                configuration.getKubeApiServerIntegratorRefreshIntervalMs()
+        );
+    }
+
+    private SharedIndexInformer<V1OpportunisticResource> createOpportunisticResourceInformer(SharedInformerFactory sharedInformerFactory) {
+        // TODO(fabio): enhance the kube client to support custom JSON deserialization options
+        return sharedInformerFactory.sharedIndexInformerFor(
+                this::listOpportunisticResourcesCall,
+                V1OpportunisticResource.class,
+                V1OpportunisticResourceList.class,
+                configuration.getKubeOpportunisticRefreshIntervalMs()
+        );
+    }
+
+    private Call listOpportunisticResourcesCall(CallGeneratorParams params) {
+        try {
```

Comment:
probably safe to remove this try catch now and the `return null` below

Reply:
This method throws checked exception. Wrapping it with `IllegalStateException`